src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -596,37 +597,32 @@ public void stop(Throwable cause) throws Exception {
      * @return the computer
      * @throws AbortException in case of error.
      */
-    private @CheckForNull Computer getComputer() throws AbortException {
+    private @Nonnull Computer getComputer() throws AbortException {
+        if (computer != null) {
+            return computer;
+        }
+
         String node = null;
-        if (computer == null) {
-            Jenkins j = Jenkins.getInstance();
-            if (j == null) {
-                LOGGER.fine("Jenkins is not running");
-                return null;
-            }
-            for (Computer c : j.getComputers()) {
-                if (c.getChannel() == launcher.getChannel()) {
-                    node = c.getName();
-                    break;
-                }
-            }
+        Jenkins j = Jenkins.getActiveInstance();
 
-            if (node == null) {
-                LOGGER.log(Level.FINE, "could not find comuter for this job");
-                return null;
+        for (Computer c : j.getComputers()) {
+            if (c.getChannel() == launcher.getChannel()) {
+                node = c.getName();
+                break;
             }
+        }
 
-            computer = j.getComputer(node);
-            if (computer == null) {
-                LOGGER.log(Level.FINE, "no such computer {0}", node);
-                return null;
-            }
+        if (node == null) {
+            throw new AbortException("Could not find computer for the job");
+        }
 
-            if (computer.isOffline()) {
-                LOGGER.log(Level.FINE, "{0} is offline", node);
-                return null;
-            }
+        computer = j.getComputer(node);
+        if (computer == null) {
+            throw new AbortException("No such computer " + node);
+        }
 
+        if (computer.isOffline()) {

```

Comment:
should this abort?

if the computer is offline after the workflow has entered the `node` block then you should still be able to use it.  (if it is temporarily offline).  if it is offline as the master has lost the connection then should this abort?  pipelines are supposed to be (mostly) tolerant to slave disconnects...